generate.go

```diff
@@ -154,9 +159,110 @@ func modify(spec *specs.LinuxSpec, rspec *specs.LinuxRuntimeSpec, context *cli.C
 	if err := addIDMappings(spec, rspec, context); err != nil {
 		return err
 	}
+	if err := addSeccomp(spec, rspec, context); err != nil {
+		return err
+	}
 
 	return nil
 }
+func addSeccomp(spec *specs.LinuxSpec, rspec *specs.LinuxRuntimeSpec, context *cli.Context) error {
+	//set the DefaultAction of seccomp
+	sd := context.String("seccomp-default")
+	switch sd {
+	case "":
+	case "SCMP_ACT_KILL":
+	case "SCMP_ACT_TRAP":
+	case "SCMP_ACT_ERRNO":
+	case "SCMP_ACT_TRACE":
+	case "SCMP_ACT_ALLOW":
+	default:
+		return fmt.Errorf("seccomp-default must be empty or one of SCMP_ACT_KILL|SCMP_ACT_TRAP|SCMP_ACT_ERRNO|SCMP_ACT_TRACE|SCMP_ACT_ALLOW")
+	}
+	rspec.Linux.Seccomp.DefaultAction = specs.Action(sd)
+	//add the additional architectures permitted to be used for system calls
+	for _, archs := range context.StringSlice("seccomp-arch") {
+		switch archs {
+		case "":
+		case "SCMP_ARCH_X86":
+		case "SCMP_ARCH_X86_64":
+		case "SCMP_ARCH_X32":
+		case "SCMP_ARCH_ARM":
+		case "SCMP_ARCH_AARCH64":
+		case "SCMP_ARCH_MIPS":
+		case "SCMP_ARCH_MIPS64":
+		case "SCMP_ARCH_MIPS64N32":
+		case "SCMP_ARCH_MIPSEL":
+		case "SCMP_ARCH_MIPSEL64":
+		case "SCMP_ARCH_MIPSEL64N32":
+		default:
+			return fmt.Errorf("seccomp-arch must be empty or one of SCMP_ARCH_X86|SCMP_ARCH_X86_64|SCMP_ARCH_X32|SCMP_ARCH_ARM|SCMP_ARCH_AARCH64SCMP_ARCH_MIPS|SCMP_ARCH_MIPS64|SCMP_ARCH_MIPS64N32|SCMP_ARCH_MIPSEL|SCMP_ARCH_MIPSEL64|SCMP_ARCH_MIPSEL64N32")
+		}
+		rspec.Linux.Seccomp.Architectures = append(rspec.Linux.Seccomp.Architectures, specs.Arch(archs))
+	}
+	// set syscall restrict in Seccomp
+	// the format of input syscall string is Name:Action:Args[1],Args[2],...,Args[n]
+	// the format of Args string is Index/Value/ValueTwo/Operator,and is parsed by function parseArgs()
+	for _, syscalls := range context.StringSlice("seccomp-syscalls") {
+		syscall := strings.Split(syscalls, ":")
+		if len(syscall) == 3 {
+			name := syscall[0]
+			switch syscall[1] {
+			case "":
+			case "SCMP_ACT_KILL":
+			case "SCMP_ACT_TRAP":
+			case "SCMP_ACT_ERRNO":
+			case "SCMP_ACT_TRACE":
+			case "SCMP_ACT_ALLOW":
+			default:
+				return fmt.Errorf("seccomp-sysctl action must be empty or one of SCMP_ACT_KILL|SCMP_ACT_TRAP|SCMP_ACT_ERRNO|SCMP_ACT_TRACE|SCMP_ACT_ALLOW")
+			}
+			action := specs.Action(syscall[1])
+			Args, err := parseArgs(syscall[2])
+			if err != nil {
+				return err
+			}
+			syscallstruct := specs.Syscall{name, action, Args}
+			rspec.Linux.Seccomp.Syscalls = append(rspec.Linux.Seccomp.Syscalls, &syscallstruct)
+		} else {
+			return fmt.Errorf("seccomp sysctl must consist of 3 parameters")
```

Comment:
We shouldn't add a commit for fixing the typo. The original commit should be fixed.